images/virtualization-artifact/pkg/controller/vi/internal/source/upload.go

```diff
@@ -31,28 +35,246 @@ import (
 	"github.com/deckhouse/virtualization-controller/pkg/dvcr"
 	"github.com/deckhouse/virtualization-controller/pkg/logger"
 	virtv2 "github.com/deckhouse/virtualization/api/core/v1alpha2"
+	"github.com/deckhouse/virtualization/api/core/v1alpha2/vdcondition"
 	"github.com/deckhouse/virtualization/api/core/v1alpha2/vicondition"
 )
 
+const uploadDataSource = "registry"
+
 type UploadDataSource struct {
 	statService     Stat
 	uploaderService Uploader
 	dvcrSettings    *dvcr.Settings
+	diskService     *service.DiskService
 }
 
 func NewUploadDataSource(
 	statService Stat,
 	uploaderService Uploader,
 	dvcrSettings *dvcr.Settings,
+	diskService *service.DiskService,
 ) *UploadDataSource {
 	return &UploadDataSource{
 		statService:     statService,
 		uploaderService: uploaderService,
 		dvcrSettings:    dvcrSettings,
+		diskService:     diskService,
 	}
 }
 
-func (ds UploadDataSource) Sync(ctx context.Context, vi *virtv2.VirtualImage) (bool, error) {
+func (ds UploadDataSource) StoreToPVC(ctx context.Context, vi *virtv2.VirtualImage) (bool, error) {
+	log, ctx := logger.GetDataSourceContext(ctx, uploadDataSource)
```

Comment:
the value of uploadDataSource is "registry"

Reply:
Fixed.